File: gherkin/dart/lib/src/messages/Envelope.dart
import 'package:gherkin/core.dart';
import 'package:gherkin/src/ast/GherkinDocument.dart';
import 'package:gherkin/src/messages/Source.dart';
import 'package:gherkin/src/pickles/Pickle.dart';

class Envelope implements INullSafetyObject
{
  static final empty = _InvalidEnvelope();

  GherkinDocument gherkinDocument = GherkinDocument.empty;

  Source source = Source.empty;

  Pickle pickle = Pickle.empty;

  @override
  bool get isEmpty => false;

  @override
  bool get isNotEmpty => !isEmpty;
}

// Convenience implementation of an invalid [Envelope] instance.
class _InvalidEnvelope
    extends Envelope
{
  @override
  bool get isEmpty => true;
}

File: gherkin/dotnet/Gherkin/CucumberMessages/Types/Envelope.cs
using System;
using System.Runtime.Serialization;

namespace Gherkin.CucumberMessages.Types
{
    public class Envelope
    {
        [DataMember(Name = "gherkinDocument")]
        public GherkinDocument GherkinDocument { get; set; }

        [DataMember(Name = "source")]
        public Source Source { get; set; }

        [DataMember(Name = "pickle")]
        public Pickle Pickle { get; set; }

        [DataMember(Name = "parseError")]
        public ParseError ParseError { get; set; }
    }
}
